managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3080,12 +3080,11 @@ private void checkAddTimeout() {
         OpAddEntry opAddEntry = pendingAddEntries.peek();
         if (opAddEntry != null) {
             boolean isTimedOut = opAddEntry.lastInitTime != -1
-                    && TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - opAddEntry.lastInitTime) >= timeoutSec
-                    && opAddEntry.completed == FALSE;
+                    && TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - opAddEntry.lastInitTime) >= timeoutSec;
```

Comment:
Don't understand why we removed `opAddEntry.completed == FALSE`

Reply:
we need atomic callback handler and want to make sure that `OpAddEntry` callback completes atomically. So, we replaced boolean `completed` with `ctxForCallbackComplete` so, we can update it atomically and only one thread (bk-client/timeout-task) succeed to process the callback.